pkg/splunk/client/minioclient.go

```diff
@@ -0,0 +1,102 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// blank assignment to verify that MinioClient implements S3Client
+var _ S3Client = &MinioClient{}
+
+// MinioClient is a client to implement S3 specific APIs
+type MinioClient struct {
+	BucketName        string
+	S3AccessKeyID     string
+	S3SecretAccessKey string
+	Prefix            string
+	StartAfter        string
+	Endpoint          string
+}
+
+// NewMinioClient returns an Minio client
+func NewMinioClient(bucketName string, accessKeyID string, secretAccessKey string, prefix string, startAfter string, endpoint string) S3Client {
+	return &MinioClient{
+		BucketName:        bucketName,
+		S3AccessKeyID:     accessKeyID,
+		S3SecretAccessKey: secretAccessKey,
+		Prefix:            prefix,
+		StartAfter:        startAfter,
+		Endpoint:          endpoint,
+	}
+}
+
+//RegisterMinioClient will add the corresponding function pointer to the map
+func RegisterMinioClient() {
+	S3Clients["minio"] = NewMinioClient
+}
+
+// GetAppsList get the list of apps from remote storage
+func (client *MinioClient) GetAppsList() (S3Response, error) {
+	scopedLog := log.WithName("GetAppsList")
+
+	scopedLog.Info("Getting Apps list", " S3 Bucket", client.BucketName)
+	s3Resp := S3Response{}
+	appS3Endpoint := client.Endpoint
+
+	useSSL := true
+	if strings.HasPrefix(appS3Endpoint, "http://") {
+		useSSL = false
+		appS3Endpoint = strings.TrimPrefix(appS3Endpoint, "http://")
+	} else if strings.HasPrefix(appS3Endpoint, "https://") {
+		appS3Endpoint = strings.TrimPrefix(appS3Endpoint, "https://")
+	} else {
+		// Unsupported endpoint
+		scopedLog.Info("Unsupported endpoint for Minio S3 client", "appS3Endpoint", appS3Endpoint)
+		return s3Resp, nil
+	}
+
+	// New returns an Minio compatible client object. API compatibility (v2 or v4) is automatically
+	// determined based on the Endpoint value.
+	scopedLog.Info("Connecting to Minio S3 for apps", "appS3Endpoint", appS3Endpoint, "appBkt", client.BucketName)
+	s3Client, err := minio.New(appS3Endpoint, &minio.Options{
```

Comment:
Can you create a new function just to get this s3Client, something like `InitMinioClientSession`? This will be helpful in changes I am making for the new framework that covers UTs too.

Reply:
Sure, I've moved this to a separate function.